### DISCUSSION TITLE
[Question] How can I set up limits for number of records

### DISCUSSION THREAD

**Discussion by SyntaxConqueror** [Laravel-Backpack/community-forum#1165] [Q&A (Help)] [answered]
I want to set up limits for number of records in my model crud page. 

For example I have Users and I want to show only 10 of them.
How can I do that?

It would be better if I could do limits for some parameter. For example top 10 users by their marks

**Comment by siberfx** [comment]
you can take a look of;
``php 
CRUD::addClause('orderBy', 'columnName', 'condition')`
``
example usage :

``php 
CRUD::addClause('orderBy', 'top_level', 'asc')
``

**Reply by SyntaxConqueror** [reply]
I need to view a proper amount of Users. Not order them by some condition
For example in Database I have 100 Users - I need to list only 10 of them who has the best marks

**Comment by jcastroa87** [comment]
Hello @SyntaxConqueror 

Im not fully sure about what kind of limit you want to use, but to limit the list records you can change on config/backpack/crud.php
```
'defaultPageLength' => 10,
'pageLengthMenu' => [[10, 25, 50, 100, -1], [10, 25, 50, 100, 'backpack::crud.all']],
```

If you need to limit results, as @siberfx say, you can use
```
CRUD::limit();
```

Let me know if that work for you.

Cheers.

**Reply by siberfx** [reply]
he can filter and sort by that, he wants to limit the results returning from Model if I understand correctly. so addClause is a way of he can determine and limit

**Reply by SyntaxConqueror** [reply]
I need to view a proper amount of Users. Page length is not a solution.
For example in Database I have 100 Users - I need to list only 10 of them who has the best marks

**Reply by pxpm** [reply]
@SyntaxConqueror it depends on how you have users and marks stored. 

If there is a `score` column in your user table you can just `CRUD::orderBy('score', 'desc')->limit(10)`. If you need to write any complex query, you can just:
```php
$query = User::with('grades')->sum()->avg()->whatever();
CRUD::setQuery($query);
```

Hope that helps. 

Cheers

**Reply by SyntaxConqueror** [reply]
> @SyntaxConqueror it depends on how you have users and marks stored.
> 
> If there is a `score` column in your user table you can just `CRUD::orderBy('score', 'desc')->limit(10)`. If you need to write any complex query, you can just:
> 
> ```
> $query = User::with('grades')->sum()->avg()->whatever();
> CRUD::setQuery($query);
> ```
> 
> Hope that helps.
> 
> Cheers

It didn't help. I still have all of Users listed

**Reply by siberfx** [reply]
why dont you just use filters instead ? I think you should share your query which you have just used but didnt work. we can see where is your problem exactly

**Comment by SyntaxConqueror** [comment, marked as answer]
Here is my answer. Thanks for user @siberfx who helped me with the idea.

```
CRUD::filter('top10')
            ->type('simple')
            ->whenActive(function() {
                $fourWeeksAgo = Carbon::now()->subWeeks(4);
                $now = Carbon::now()->endOfWeek();

                $limitValue = \App\Models\Child::withSum(['academicMarks as totalMarks' => function ($query) use ($fourWeeksAgo, $now) {
                    $query->whereBetween('date', [$fourWeeksAgo, $now]);
                }], 'mark')
                    ->having('totalMarks', '>=', 0)
                    ->orderBy('totalMarks', 'desc')
                    ->skip(9)
                    ->take(1)
                    ->value('totalMarks');

                $limitValue = $limitValue ?? 0;
                
                return $this->crud->query
                    ->whereHas('academicMarks', function ($query) use ($fourWeeksAgo, $now) {
                        $query->whereBetween('date', [$fourWeeksAgo, $now]);
                    })
                    ->withSum(['academicMarks as totalMarks' => function ($query) use ($fourWeeksAgo, $now) {
                        $query->whereBetween('date', [$fourWeeksAgo, $now]);
                    }], 'mark')
                    ->having('totalMarks', '>=', $limitValue);
            });
```
